src/controller/markdown.controller.ts

```diff
@@ -0,0 +1,133 @@
+          acc += chunk;
+          index !== -1 ? stream.close() : (pos += chunk.length);
+        })
+        .on('close', () => {
+          resolve(acc.slice(0, pos + index));
+        })
+        .on('error', (err) => {
+          reject(err);
+        });
+    });
+  }
+  private static async buildNewsHeadMapping(domain: MARKDOWN_DOMAIN): Promise<[string, string][]> {
+    const newsFiles = MarkdownController.getAvailableFiles(domain);
+    const mapping_p: Promise<[string, string]>[] = newsFiles.map(
+      async (fileName: string) => {
+        const pair: Promise<[string, string]> = MarkdownController.firstLineFs(
+          MarkdownController.getPathForDomain(domain) + '/' + fileName
+        ).then((head) => [fileName, head]);
+        return pair;
+      }
+    );
+    return Promise.all(mapping_p);
+  }
+
+
+  static async availableFilesRequest(req: Request, res: Response, domain: MARKDOWN_DOMAIN): Promise<void> {
+    const payload = MarkdownController.getAvailableFiles(domain);
+    new APIResponse(res, 200, payload).send();
+  }
+  static async fileRequest(req: Request, res: Response, domain: MARKDOWN_DOMAIN): Promise<void> {
+    if (req.params.md === undefined || req.params.md === null)
+      new APIResponse(res, 400, {}, ['Undefined news parameter']).send();
+
+    const path = `${MarkdownController.getPathForDomain(domain)}/${req.params.md}`;
+
+    try {
+      if (fs.existsSync(path)) {
+        const fileContents: Buffer = fs.readFileSync(path);
+        console.log("file loaded", fileContents.toString());
+        const parsedHTML: string = marked.parse(fileContents.toString());
+        console.log("file parsed", parsedHTML);
+        res.set('Content-Type', MIMETYPES.md);
+        res.send(parsedHTML);
+        console.log("file sent");
+      } else {
+        new APIResponse(res, 404, {}, ['File not found']).send();
+      }
+    } catch (e) {
+      console.error("fs error", e);
+
+      new APIResponse(res, 500, {}, ['File access error']).send();
+    }
+  }
+  static async mediaRequest(req: Request, res: Response, domain: MARKDOWN_DOMAIN): Promise<void> {
+    const path = MarkdownController.getPathForDomain(domain) + '/' + req.params.fileName;
+    if (path !== undefined && path !== null) {
+      try {
+        if (fs.existsSync(path)) {
+          const stream = fs.createReadStream(path);
+          stream.on('open', () => {
+            stream.pipe(res);
+          });
+        } else {
+          new APIResponse(res, 404, {}, ['File not found']).send();
+        }
+      } catch (e) {
+        new APIResponse(res, 500, {}, ['File access error']).send();
+      }
+    } else {
+      new APIResponse(res, 400, {}, ['Undefined fileName parameter']).send();
+    }
+  }
+  static async newsHeadRequest(req: Request, res: Response, domain: MARKDOWN_DOMAIN): Promise<void> {
+    const path = MarkdownController.getPathForDomain(domain) + '/' + req.params.md;
+    if (path !== undefined && path !== null) {
```

Comment:
```suggestion
    if (path) { 
```

Ich meine damit solltest du sogar beides direkt abgedeckt haben. Alternativ:

```suggestion
    if (!!path) {
```

Reply:
was wenn path = false?
dann evaluiert 'path' zu false und 'path!==undefined&&path!==null' zu true